### PR TITLE
Migrate Drawer, ErrorDisplay, and FeaturedJobItem components to typescript

### PR DESCRIPTION
**Pull request by recondesigns**
# Description of changes

Migrated the following components to Typescript, removed prop-types, and added a props type:

- `Drawer`
- `ErrorDisplay`
- `FeaturedJobItem`

*Migrating components included all test and storybook files for each.*

# Issue Resolved

Fixes #NA

#### All new `.tsx` files

Screenshot 2024-03-04 at 12 45 49 PM

### CODE REVIEW

File: components/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+function Drawer({ children, className, isVisible = false }: DrawerPropsType) {
+  return (
+    <div
+      className={classNames(
+        className,
+        'hidden lg:block transition-all ease-in-out duration-1000 fixed top-0 bottom-0  overflow-hidden width-full z-[2]',
+        {
+          '-left-0': isVisible,
+          '-left-[100%]': !isVisible,
```
Comment (kylemh) [inline components/Drawer/Drawer.tsx:27]:
```suggestion
          '-left-full': !isVisible,
```
i know this wasn't you

File: components/Drawer/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   */
+  isVisible?: boolean;
+};
+
+function Drawer({ children, className, isVisible = false }: DrawerPropsType) {
+  return (
+    <div
+      className={classNames(
+        className,
+        'hidden lg:block transition-all ease-in-out duration-1000 fixed top-0 bottom-0  overflow-hidden width-full z-[2]',
```
Comment (kylemh) [inline components/Drawer/Drawer.tsx:24]:
```suggestion
        'hidden lg:block transition-all ease-in-out duration-1000 fixed top-0 bottom-0 overflow-hidden width-full z-[2]',
```
same

File: components/Drawer/__stories__/Drawer.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// };
+
+// const Template = arguments_ => <Drawer {...arguments_} />;
+
+// // Default Drawer supplied with only required args
+// export const Default = Template.bind({});
+// Default.args = {
+// children: 'Drawer content will only display on display size of Tablet or smaller',
+// isVisible: true,
+// };
```
Comment (kylemh) [inline components/Drawer/__stories__/Drawer.stories.tsx:46]: 🚧

**Review by kylemh — APPROVED** [review]

File: components/Drawer/Drawer.tsx (hunk 2 above)
Comment (recondesigns) [reply components/Drawer/Drawer.tsx:24]: There doesn't appear to be a change in the suggestion. Am I missing a tiny detail?

Comment (kylemh) [reply components/Drawer/Drawer.tsx:24]: there are 2 spaces between `bottom-0` and `overflow-hidden`. Perhaps you need to lower the contrast on your monitor?

Screenshot 2024-03-05 at 3 05 09 PM

**Review by kylemh — APPROVED** [review]

File: components/Drawer/Drawer.tsx (hunk 2 above)
Comment (recondesigns) [reply components/Drawer/Drawer.tsx:24]: Ahhhh...  I actually did see it, but didn't know what it was. This is merged, should I make that change in one of the other branches I am working?

**Pull request merged**